scipy/stats/_continuous_distns.py

```diff
@@ -5439,18 +5439,21 @@ def f3(q, h, k):
                            default=np.nan)
 
     def _stats(self, h, k):
-        if h >= 0 and k >= 0:
-            maxr = 5
-        elif h < 0 and k >= 0:
-            maxr = int(-1.0/h*k)
-        elif k < 0:
-            maxr = int(-1.0/k)
-        else:
-            maxr = 5
-
-        outputs = [None if r < maxr else np.nan for r in range(1, 5)]
+        maxr = np.where(np.logical_and(h >= 0, k >= 0), 5, np.nan)
+        maxr = np.where(np.logical_and(h < 0, k >= 0, np.isnan(maxr)),
+                        (-1.0/h*k).astype(int), maxr)
+        maxr = np.where(np.logical_and(k < 0, np.isnan(maxr)),
+                        (-1.0/k).astype(int), maxr)
+        maxr = np.where(np.isnan(maxr), 5, maxr)
+
+        outputs = [None if np.any(r < maxr)  else np.nan for r in range(1, 5)]
         return outputs[:]
 
+    def _mom1_sc(self, m, *args):
+        if np.any(m >= args[0]):
```

Comment:
Here, `args[0]` is `h`. This is not the condition you want; you need `maxr`. You'll need to recalculate it.
Everything is `nan` now, and that's why the test is passing.